doc/reference/nhibernate_caches.md

```diff
@@ -0,0 +1,715 @@
+# NHibernate.Caches
+
+> **NHibernate.Caches namespace contains several second-level cache
+> providers for NHibernate..**
+> 
+> A cache is a place where entities are kept after being loaded from the
+> database; once cached, they can be retrieved without going to the
+> database. This means that they are faster to (re)load.
+> 
+> An NHibernate session has an internal (first-level) cache where it
+> keeps its entities. There is no sharing between these caches - a
+> first-level cache belongs to a given session and is destroyed with it.
+> NHibernate provides a *second-level cache* system; it works at the
+> session factory level. A second-level cache is shared by all sessions
+> created by the same session factory.
+> 
+> An important point is that the second-level cache *does not* cache
+> instances of the object type being cached; instead it caches the
+> individual values of the properties of that object. This provides two
+> benefits. One, NHibernate doesn't have to worry that your client code
+> will manipulate the objects in a way that will disrupt the cache. Two,
+> the relationships and associations do not become stale, and are easy
+> to keep up-to-date because they are simply identifiers. The cache is
+> not a tree of objects but rather a map of arrays.
+> 
+> With the *session-per-request* model, a high number of sessions can
+> concurrently access the same entity without hitting the database each
+> time; hence the performance gain.
+> 
+> Depending on the chosen cache provider, the second level cache may be
+> actually shared between different session factories. If you need to
+> avoid this for some session factories, configure each of them with a
+> different `cache.region_prefix`. See [???](#configuration-optional).
```

Comment:
As reported by Roman, internal links are broken.

Reply:
I went through all of these and I think I have fixed all of them. I will still do a manual check.